src/apps.rs

```diff
@@ -149,9 +148,7 @@ pub fn add_trust_anchor(
             StatusCode::OK => {
                 let app_obj = r.text().unwrap_or_else(|_| "{}".to_string());
                 let mut app_obj: Value = serde_json::from_str(&app_obj)?;
-                app_obj.merge(json!({
-                    "spec": trust::create_trust_anchor(app, keyout, days)?
-                }));
+                app_obj["spec"] = trust::create_trust_anchor(app, keyout, days)?;
```

Comment:
App can have arbitrary data in the `spec` object, you can't overwrite the whole thing.
Please have `create_trust_anchor` function returns only the trustAnchor section of the object and merge it with the rest.

```suggestion
                app_obj["spec"]["trustAnchors"] = trust::create_trust_anchor(app, keyout, days)?;
```